app/models/interface.rb

```diff
@@ -0,0 +1,24 @@
+class Interface < ApplicationRecord
+  enum interface_type: { discord_guild: 0, web: 1, api: 2 }
+
+  has_many :messages, dependent: :destroy
+
+  validates :interface_type, presence: true
+  validates :external_id, presence: true, if: :requires_external_id?
+
+  class << self
+    def web
+      find_or_create_by(interface_type: :web)
+    end
+
+    def api
+      find_or_create_by(interface_type: :api)
+    end
+  end
+
```

Comment:
rails already provides these methods when you define enums. Can we use those instead, please? Also please create a migration or a rake task to create these initial web interfaces, and add a custom validation to check that these two specific types are unique

Reply:
Done! We don’t need the rake task. When a message is created on the web for the first time, the singleton web interface_type record will be created

Reply:
No, that's too complex to read in the code. Let's use a rake task to seed these interfaces and add a custom validation to ensure that these sources are always unique, then in whatever places you need to find then you find, and in whatever places you need to create then you create.